core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -445,7 +458,11 @@ public Reader asReader(@Nullable String defaultDecodingCharset) throws IOExcepti
 			LOG.debug("returning InputStream {} as Reader", this::getObjectId);
 			InputStream inputStream = asInputStream();
 			try {
-				return StreamUtil.getCharsetDetectingInputStreamReader(inputStream, readerCharset);
+				BufferedReader reader = new BufferedReader(StreamUtil.getCharsetDetectingInputStreamReader(inputStream, readerCharset));
```

Comment:
Deze `getCharsetDetectingInputStreamReader` gebruikt onder water een BufferedInputStream, misschien dat het handig is deze altijd een buffered reader terug te geven?

Reply:
Daar ga ik nog even naar kijken. Want ik had al wel gezien dat daar inderdaad heel wat buffered streams over elkaar heen gelegd worden.